### PR TITLE
Pin requirements

### PR DESCRIPTION
**Pull request by allait**
requirements.txt is generated using a Python 3 virtualenv to keep the py2 backports of Python 3 stdlib packages (like `enum34` and `ipaddress`) out. This leaves those packages unpinned, but shouldn't cause any issues otherwise.

### Reorganize Makefile and requirements files

Renames requirements.txt to requirements-app.txt and requirements_for_test.txt to requirements-dev.txt.

Removes any unused dependencies or dependencies provided by other packages since those will be listed in the full requirements.txt.

Updates Makefile to add freeze-requirements and rename targets to be more consistent with other apps.

(see alphagov/digitalmarketplace-api#615 for details)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,53 +8,83 @@ else
...
 	npm install
 
-frontend_build:
+.PHONY: frontend-build
+frontend-build:
 	npm run --silent frontend-build:${GULP_ENVIRONMENT}
 
-test: show_environment frontend_build test_pep8 test_python test_javascript
+.PHONY: test
+test: show-environment test-requirements frontend-build test-pep8 test-python test-javascript
```
Comment (Wynndow) [inline Makefile]: Is it worth adding `requirements-dev` to this?

File: Makefile [hunk 2]
```diff
@@ -8,53 +8,83 @@ else
...
+.PHONY: requirements
+requirements: virtualenv test-requirements upgrade-pip requirements.txt
 	${VIRTUALENV_ROOT}/bin/pip install -r requirements.txt
 
-requirements_for_test: virtualenv upgrade_pip requirements_for_test.txt
-	${VIRTUALENV_ROOT}/bin/pip install -r requirements_for_test.txt
-
-npm_install: package.json
+.PHONY: requirements-dev
+requirements-dev: virtualenv test-requirements requirements-dev.txt
```
Comment (Wynndow) [inline Makefile]: Would `requirements-test` be a better name for this (and the requirements file)? Bit of a pain to change them all now though...

File: Makefile [hunk 3]
```diff
@@ -8,53 +8,83 @@ else
...
-test_pep8: virtualenv
+.PHONY: test-pep8
+test-pep8: virtualenv
 	${VIRTUALENV_ROOT}/bin/pep8 .
 
-test_python: virtualenv
-	${VIRTUALENV_ROOT}/bin/py.test ${PYTEST_ARGS}
+.PHONY: test-python
+test-python: virtualenv
+	${VIRTUALENV_ROOT}/bin/py.test ${PYtest-ARGS}
```
Comment (Wynndow) [inline Makefile]: Lower case used here for the variable, but all upper in the `.travis.yml`

File: Makefile (hunk 2 above)
Comment (allait) [inline Makefile]: That's what it used to be (`requirements_for_test.txt`). The reason for the rename was to reflect the fact that this should also be used for non-test related dev dependencies - like script dependencies or other tools we use that shouldn't be installed in live environments.

File: Makefile (hunk 1 above)
Comment (allait) [inline Makefile]: We kept them as separate steps so far. We could make an argument about `make test` running pip install each time now that the versions are pinned, but it feels like a separate conversation to me.

File: Makefile (hunk 3 above)
Comment (allait) [inline Makefile]: Oops! should be fixed now

**Pull request merged**